src/common/consensus-provider/consensus-provider.service.ts

```diff
@@ -290,10 +290,19 @@ export class ConsensusProviderService {
       return cached.missed ? undefined : cached.info;
     }
 
+    const workingMode = this.config.get('WORKING_MODE');
+
     const blockInfo = await this.retryRequest<BlockInfoResponse>(
       async (apiURL: string) => this.apiGet(apiURL, this.endpoints.blockInfo(blockId)),
       {
         maxRetries: this.config.get('CL_API_GET_BLOCK_INFO_MAX_RETRIES'),
+        useFallbackOnResolved: (r) => {
```

Comment:
let's investigate what else has `finalized` field (`headers` endpoint, at least) and add the callback there as well

Reply:
Yes, that's a good idea. I've checked other endpoints that are used by the Consensus Provider service and made sure that almost all of them have the `finalized` field in their responses (except the `eth/v1/validator/duties/proposer/${epoch}` endpoint).

I've added the callback to the `getLatestBlockHeader` and `getSyncCommitteeInfo` methods. Also, I've added checking that these methods are not called for the head block.

It seemed logical to add this callback also to other methods of the Consensus Provider, but I didn't do that for a number of reasons.

I didn't add the callback to the `getBlockHeader` method because it is used by the `high-reward-validators` task (from the Duty service) to get the canonical proposer duties. These duties are fetched for the head epoch and the `getBlockHeader` method of the Consensus Provider is called in this process for block IDs that are not finalized yet (even if the app is working in the "finalized" mode).
So, checking the `finalized` field from the node response in this method doesn't make sense, because it might be normally called for finalized and non-finalized blocks.

I didn't add the callback to the `getState` and `getAttestationCommitteesInfo` methods because they use the `apiGetStream` method to fetch data from the node. The current version of the `retryRequest` method doesn't support checking data in the node response in the `useFallbackOnRejected` callback if data were fetched using the `apiGetStream` method. It doesn't look important to modify the `retryRequest` method and add support for data streams to its callbacks right now. The `getState` and `getAttestationCommitteesInfo` methods are called less often than other methods where the `useFallbackOnRejected` callback was added. I hope that having this callback in the methods where it was added should be enough to catch a non-finalized node state if the issue with a node happens.